test-sbt/shared/src/main/scala/zio/test/sbt/BaseTestTask.scala

```diff
@@ -41,11 +41,14 @@ abstract class BaseTestTask(val taskDef: TaskDef, testClassLoader: ClassLoader)
         }
       }
 
-    for {
-      result   <- spec.runWith(TestReporter.silent)
-      rendered = DefaultTestReporter.render(result.mapLabel(_.toString))
-      _        <- reportResults(rendered)
-    } yield ()
+    val testReporter: TestReporter[spec.Label] = result => {
+      val rendered = DefaultTestReporter.render(result.mapLabel(_.toString))
```

Comment:
What would be nice is if it could use `spec.runner.reporter` instead of overriding it with the default one, to respect what's defined by the user in the test. Maybe users might change the way their tests are reported and it's pity to lose it when using sbt.

We could change `TestReporter` to be `ExecutedSpec[L] => URIO[Console, Unit]`, remove the `Console` passed in `apply` and instead pass it when we `run`. When we run it from sbt, we would pass an implementation of `Console` that prints to sbt (`log.info`), otherwise we would just pass `Console.Live`.

Reply:
Well `Console` doesn't map exactly to the sbt logger API.
First there is no way to implement a `putStr` - `log.info()` is `putStrLn`.
I guess I could accumulate the calls to `putStr` until I see `putStrLn` but this feels unnecessarily complicated...  
Also there is nothing for `getStrLn`. Of course there is no reason for a `TestReporter` to use it, but still, it doesn't feel very clean...

Reply:
Maybe we extend `Console.Live` and only override `putStrLn`?
That way you get the expected behavior and only when appending a full line you forward it to sbt.

Reply:
> Well Console doesn't map exactly to the sbt logger API.

Does anything actually pins us to using Console instead of creating a separate interface?

Reply:
Yep we can just use a new one 👍

Reply:
Ok, I changed the definition of `TestReporter` to:
```scala
 type TestReporter[-L] = ExecutedSpec[L] => URIO[TestLogger, Unit]
```
Where `TestLogger.Service` looks like this:
```scala
trait Service {
    def logLine(line: String): UIO[Unit]
 }
```